### ISSUE TITLE
The name `AssemblyCommitIds` does not exist in the current context.

### ISSUE THREAD

**Issue by m-rienstra** [libgit2/libgit2sharp#1512]
When I try to compile the new LibGit2Sharp (retrieved from GitHub) I get the error:
***The name `AssemblyCommitIds` does not exist in the current context.***
Trying to compile it using Visual Studio 2017 (Community Edition), the reason for compilation is that I'm trying to create a custom Git "EditorWindow" for Unity 2017.3b.
Thanks in advance!

**Comment by m-rienstra** [comment]
I just realized that it might be useful to know that the error occurs in `Version.cs` and is caused by 2 string properties:

*Line 40*
```
public virtual string LibGit2CommitSha => RetrieveAbbrevShaFrom(AssemblyCommitIds.LibGit2CommitSha);
```

*Line 45*
```
public virtual string LibGit2SharpCommitSha => RetrieveAbbrevShaFrom(AssemblyCommitIds.LibGit2SharpCommitSha);
```

**Comment by ethomson** [comment]
Out of curiosity, does it work if you run `dotnet build` on the command line?

**Comment by m-rienstra** [comment]
That gives the same output, the issue is that `AssemblyCommitIds` is not present in any of the scripts.

**Comment by bording** [comment]
@m-rienstra AssemblyCommitIds is a class generated by MSBuild targets: https://github.com/libgit2/libgit2sharp/blob/master/LibGit2Sharp/CodeGenerator.targets#L80

How did you get the copy of the code you're trying to build? Did you check it out via git?

**Comment by m-rienstra** [comment]
I used git to clone the repository to a local drive, then I used visual studio to open it and tried to build it

**Comment by bording** [comment]
If you look in the LibGit2Sharp obj folder, do you see an `AssemblyCommitIds.g.cs` file in there? What about `NativeDllName.g.cs` or `UniqueIdentifier.g.cs`?

**Comment by m-rienstra** [comment]
This is what I have in the folder:
![image](https://user-images.githubusercontent.com/27005081/33036336-579d88f6-ce2e-11e7-928c-29a4c5c6d0cf.png)
There's no `AssemblyCommitIds.g.cs`

**Comment by bording** [comment]
@m-rienstra The only reason I can think of that might cause that to happen would be if you were building the code from outside of a cloned copy, meaning you don't have a .git folder.

Can you install http://msbuildlog.com/ and build the solution with that and then attach copy of the log file to the issue ? That might help me understand better why the target apparently isn't running.

**Comment by m-rienstra** [comment]
@bording Here's the error log
https://expirebox.com/download/91ab05cf76f2f2b08f33d29514e6a7e5.html

**Comment by bording** [comment]
@m-rienstra Looking at the log file, it appears that my original suspicion is correct. The reason you're seeing the error is that the `GitCommitId` property is not being set by the Nerdbank,GitVersioning's GetBuildVersion target.

I was just able to reproduce this by deleting my .git folder, making my copy of the code not be a git repository.

Looking at the log, I see your code is in the following path: `C:\Users\markr\Documents\Workspace\External\libgit2sharp-master`

If you look in that folder, I suspect you'll find that you don't have a .git folder, meaning that isn't actually a git repo.

**Comment by bording** [comment]
@m-rienstra FYI, I've just added a change to #1511 that will let the GenerateAssemblyCommitIdsCs target handle the scenario of trying to build the project when not in a git repo. 

If you don't want to wait for that to be merged, you could try checking out that branch and see how that works for you.

**Comment by ethomson** [comment]
Fixed via #1511
